Source/Orts.Simulation/Common/Scripting/TrainControlSystem.cs

```diff
@@ -560,4 +581,12 @@ public enum MonitoringStatus
         /// </summary>
         Intervention,
     }
+
+    public struct SignalFeatures
+    {
+        public string MainHeadSignalTypeName;
+        public Aspect Aspect;
+        public float DistanceM;
+        public float SpeedLimit;
```

Comment:
Another thought: Not all signals impose speed limits. Use a `float?` type so you don't have to ambiguously code this condition with `0` or `float.MaxValue`.

Also, is it possible for signals not to have aspects?

Reply:
I don't think it is desirable to expect from the script writers to do null-checks, so I would not change it to "float?". The aspects enum has a "none" member.

Reply:
The type checker will actually throw an error if you use a `float?` in a location that requires a plain old `float` and remind you to downcast with `.Value` or `??`. Optionals are much more sophisticated than `if (x != null)` and it would be a shame not to take advantage of them.

Reply:
Er, that's ["nullable reference type"](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/nullable-reference-types) in C# speak. It's a potential null pointer that forces you to deal with the null pointer condition.

Reply:
I will leave things as they are.

Reply:
Ok, so how are you coding signals without speed limits? This needs to be documented, too.

Reply:
Correct me if I'm wrong, but I believe signal objects without speed limits have `.actual_speed = -1 m/s`.

Reply:
Yes, it's the rule used in the signalling code.

Reply:
I would have preferred nullable type too. It's cleaner than checking for negative values.